Spigot-API-Patches/0005-Adventure.patch

```diff
@@ -0,0 +1,2059 @@
++    @Deprecated // Paper
+     public String getCaption() {
+-        return caption;
++        return this.caption == null ? null : net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(this.caption); // Paper
+     }
+ 
+     /**
+      * Sets the caption on this cursor.
+      *
+      * @param caption new caption
++     * @deprecated in favour of {@link #caption(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Paper
+     public void setCaption(@Nullable String caption) {
+-        this.caption = caption;
++        this.caption = caption == null ? null : net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(caption); // Paper
+     }
+ 
+     /**
+diff --git a/src/main/java/org/bukkit/map/MapCursorCollection.java b/src/main/java/org/bukkit/map/MapCursorCollection.java
+index 4dba721aefe4fc6699b3b4bfa7ecb0b19c2a2a1a..01dec2c877df58c9dc22445e8b1f9ce2e53066da 100644
+--- a/src/main/java/org/bukkit/map/MapCursorCollection.java
++++ b/src/main/java/org/bukkit/map/MapCursorCollection.java
+@@ -117,4 +117,22 @@ public final class MapCursorCollection {
+     public MapCursor addCursor(int x, int y, byte direction, byte type, boolean visible, @Nullable String caption) {
+         return addCursor(new MapCursor((byte) x, (byte) y, direction, type, visible, caption));
+     }
++    // Paper start
++    /**
++     * Add a cursor to the collection.
++     *
++     * @param x The x coordinate, from -128 to 127.
++     * @param y The y coordinate, from -128 to 127.
++     * @param direction The facing of the cursor, from 0 to 15.
++     * @param type The type (color/style) of the map cursor.
++     * @param visible Whether the cursor is visible.
++     * @param caption banner caption
++     * @return The newly added MapCursor.
++     * @deprecated Magic value
++     */
++    @Deprecated
++    public @NotNull MapCursor addCursor(int x, int y, byte direction, byte type, boolean visible, @Nullable net.kyori.adventure.text.Component caption) {
++        return addCursor(new MapCursor((byte) x, (byte) y, direction, type, visible, caption));
++    }
++    // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/scoreboard/Objective.java b/src/main/java/org/bukkit/scoreboard/Objective.java
+index f5cbf6df32ef169cf0f2266f7c6e9c4f771ccb7d..58bddb11fd534e7c33a4ffd7b72b055ba92c767a 100644
+--- a/src/main/java/org/bukkit/scoreboard/Objective.java
++++ b/src/main/java/org/bukkit/scoreboard/Objective.java
+@@ -19,14 +19,35 @@ public interface Objective {
+      */
+     @NotNull
+     String getName() throws IllegalStateException;
++    // Paper start
++    /**
++     * Gets the name displayed to players for this objective
++     *
++     * @return this objective's display name
++     * @throws IllegalStateException if this objective has been unregistered
++     */
++    @NotNull net.kyori.adventure.text.Component displayName() throws IllegalStateException;
```

Comment:
This applies for several places in this patch: why isn't there a universal name interface being used here? The true of power Adventure can only be yielded if basic identity methods like `displayName` or `customName` are abstracted.